build/generate-gitignores.go

```diff
@@ -21,14 +21,20 @@ import (
 
 func main() {
 	var (
-		prefix      = "gitea-gitignore"
-		url         = "https://api.github.com/repos/github/gitignore/tarball"
-		destination = ""
+		prefix         = "gitea-gitignore"
+		url            = "https://api.github.com/repos/github/gitignore/tarball"
+		githubApiToken = ""
+		destination    = ""
 	)
 
 	flag.StringVar(&destination, "dest", "options/gitignore/", "destination for the gitignores")
+	flag.StringVar(&githubApiToken, "token", "", "github api token")
 	flag.Parse()
 
+	if len(githubApiToken) > 0 {
+		url += "?token=" + githubApiToken
```

Comment:
Token being passed via query string is deprecated with github. Source: https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters

Reply:
So those links imply we could just use BASIC Auth with the token so I guess that would be the thing to do